Unable to (cross)-compile for aarch64

## Expected Behavior

(Cross)-Compile for aarch64 succeeds

## Actual Behavior

Compilation fails with error:

```sh
error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:192:9
    |
192 |         &src.as_ptr() as *const *const u8 as *const *const i8,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const *const u8`
               found raw pointer `*const *const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:245:53
    |
245 |         uniform_tex: gl.GetUniformLocation(program, tex.as_ptr() as *const i8),
    |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:246:56
    |
246 |         uniform_matrix: gl.GetUniformLocation(program, matrix.as_ptr() as *const i8),
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:247:58
    |
247 |         uniform_invert_y: gl.GetUniformLocation(program, invert_y.as_ptr() as *const i8),
    |                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:248:55
    |
248 |         uniform_alpha: gl.GetUniformLocation(program, alpha.as_ptr() as *const i8),
    |                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:249:56
    |
249 |         attrib_position: gl.GetAttribLocation(program, position.as_ptr() as *const i8),
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:250:58
    |
250 |         attrib_tex_coords: gl.GetAttribLocation(program, tex_coords.as_ptr() as *const i8),
    |                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:284:40
    |
284 |                 let p = CStr::from_ptr(ext_ptr);
    |                                        ^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:293:32
    |
293 |                 CStr::from_ptr(gl.GetString(ffi::VERSION) as *const i8)
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:298:32
    |
298 |                 CStr::from_ptr(gl.GetString(ffi::VENDOR) as *const i8)
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error[E0308]: mismatched types
   --> src/backend/renderer/gles2/mod.rs:303:32
    |
303 |                 CStr::from_ptr(gl.GetString(ffi::RENDERER) as *const i8)
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`

error: aborting due to 11 previous errors; 1 warning emitted
```

## Steps to Reproduce the Problem

  1. `rustup target add aarch64-unknown-linux-gnu`
  2. `cd anvil`
  3. (Optional) source cross-compile environment
  4. `cargo build --no-default-features --features "udev,egl,logind" --target aarch64-unknown-linux-gnu` 

## Specifications

  - Version: renderer/v4
  - Platform: linux_x68_64

--- a/src/backend/renderer/gles2/mod.rs
+++ b/src/backend/renderer/gles2/mod.rs
@@ -1,6 +1,6 @@
 //! Implementation of the rendering traits using OpenGL ES 2
 
-use std::collections::HashSet;
+use std::{collections::HashSet, os::raw::c_char};
 use std::fmt;
 use std::ffi::CStr;
 use std::ptr;
@@ -189,7 +189,7 @@ unsafe fn compile_shader(
     gl.ShaderSource(
         shader,
         1,
-        &src.as_ptr() as *const *const u8 as *const *const i8,
+        &src.as_ptr() as *const *const u8 as *const *const ffi::types::GLchar,
         &(src.len() as i32) as *const _,
     );
     gl.CompileShader(shader);
@@ -242,12 +242,12 @@ unsafe fn texture_program(gl: &ffi::Gles2, frag: &'static str) -> Result<Gles2Pr
 
     Ok(Gles2Program {
         program,
-        uniform_tex: gl.GetUniformLocation(program, tex.as_ptr() as *const i8),
-        uniform_matrix: gl.GetUniformLocation(program, matrix.as_ptr() as *const i8),
-        uniform_invert_y: gl.GetUniformLocation(program, invert_y.as_ptr() as *const i8),
-        uniform_alpha: gl.GetUniformLocation(program, alpha.as_ptr() as *const i8),
-        attrib_position: gl.GetAttribLocation(program, position.as_ptr() as *const i8),
-        attrib_tex_coords: gl.GetAttribLocation(program, tex_coords.as_ptr() as *const i8),
+        uniform_tex: gl.GetUniformLocation(program, tex.as_ptr() as *const ffi::types::GLchar),
+        uniform_matrix: gl.GetUniformLocation(program, matrix.as_ptr() as *const ffi::types::GLchar),
+        uniform_invert_y: gl.GetUniformLocation(program, invert_y.as_ptr() as *const ffi::types::GLchar),
+        uniform_alpha: gl.GetUniformLocation(program, alpha.as_ptr() as *const ffi::types::GLchar),
+        attrib_position: gl.GetAttribLocation(program, position.as_ptr() as *const ffi::types::GLchar),
+        attrib_tex_coords: gl.GetAttribLocation(program, tex_coords.as_ptr() as *const ffi::types::GLchar),
     })
 }
 
@@ -275,7 +275,7 @@ impl Gles2Renderer {
 
         let (gl, exts, logger) = {
             let gl = ffi::Gles2::load_with(|s| crate::backend::egl::get_proc_address(s) as *const _);
-            let ext_ptr = gl.GetString(ffi::EXTENSIONS) as *const i8;
+            let ext_ptr = gl.GetString(ffi::EXTENSIONS) as *const c_char;
             if ext_ptr.is_null() {
                 return Err(Gles2Error::GLFunctionLoaderError);
             }
@@ -290,17 +290,17 @@ impl Gles2Renderer {
             info!(
                 log,
                 "GL Version: {:?}",
-                CStr::from_ptr(gl.GetString(ffi::VERSION) as *const i8)
+                CStr::from_ptr(gl.GetString(ffi::VERSION) as *const c_char)
             );
             info!(
                 log,
                 "GL Vendor: {:?}",
-                CStr::from_ptr(gl.GetString(ffi::VENDOR) as *const i8)
+                CStr::from_ptr(gl.GetString(ffi::VENDOR) as *const c_char)
             );
             info!(
                 log,
                 "GL Renderer: {:?}",
-                CStr::from_ptr(gl.GetString(ffi::RENDERER) as *const i8)
+                CStr::from_ptr(gl.GetString(ffi::RENDERER) as *const c_char)
             );
             info!(log, "Supported GL Extensions: {:?}", exts);
 

